pages/_app.js

```diff
@@ -1,10 +1,18 @@
+import { useState } from "react";
 import GlobalStyle from "../styles";
 
 export default function App({ Component, pageProps }) {
-	return (
-		<>
-			<GlobalStyle />
-			<Component {...pageProps} />
-		</>
-	);
+  //Hier States
+  const [productToBuy, setProduct] = useState();
+
+  return (
+    <>
+      <GlobalStyle />
+      <Component
+        {...pageProps}
+        setProduct={setProduct}
```

Comment:
Please don't drill setter function with its own name. 

```suggestion
        handleSetProductToBuy={setProductToBuy}
```